### ISSUE TITLE
Scaffold timeout with large SQL Server db with many indexes

### ISSUE THREAD

**Issue by ErikEJ** [ErikEJ/EFCorePowerTools#498]

**Comment by ErikEJ** [comment]
@lynxwall @svengeance  I am not sure what is going on. 

Please check the %localappdata%\Microsoft\VisualStudio folder for duplicate extensions

Nuke the %temp%\efreveng folder

Could you share your efpt.config.json file (maybe privately to ejlskov at hotmail dot com )

**Comment by LynxWall** [comment]
Checked for duplicate extensions and didn't see any. I also deleted the %temp%\efreveng. Still seeing same timeout error.
Will send config.json via hotmail ... Thanks!

**Comment by ErikEJ** [comment]
I was finally able to repro, see last commit - it was error 40!

**Comment by svengeance** [comment]
@ErikEJ Unfortunately it seems that it is now coercing me to use a .dacpac now. If I check "Use EF 5.0 Preview", it gives me the regular timoue

![image](https://user-images.githubusercontent.com/6847381/92640616-8dc44100-f2ab-11ea-90c4-6ca33b3aa72f.png)


```
System.Exception: Table list error: 
System.ArgumentException: Dacpac file not found: Data Source=SSISQLPRD;Initial Catalog=Phoenix;Integrated Security=True;Asynchronous Processing=False
   at ErikEJ.EntityFrameworkCore.SqlServer.Scaffolding.SqlServerDacpacDatabaseModelFactory.Create(String dacpacPath, DatabaseModelFactoryOptions options)
   at ReverseEngineer20.TableListBuilder.GetTableDefinitions() in C:\Code\EFCorePowerTools\src\GUI\RevEng.Core\TableListBuilder.cs:line 40
   at efreveng.Program.Main(String[] args) in C:\Code\EFCorePowerTools\src\GUI\efreveng\Program.cs:line 33

   at EFCorePowerTools.Handlers.ReverseEngineer.ResultDeserializer.BuildTableResult(String output) in C:\projects\efcorepowertools\src\GUI\EFCorePowerTools\Handlers\ReverseEngineer\ResultDeserializer.cs:line 47
   at ReverseEngineer20.ReverseEngineer.EfRevEngLauncher.GetTablesInternal(String arguments) in C:\projects\efcorepowertools\src\GUI\EFCorePowerTools\Handlers\ReverseEngineer\EfRevEngLauncher.cs:line 64
   at ReverseEngineer20.ReverseEngineer.EfRevEngLauncher.GetTables(String connectionString, DatabaseType databaseType, SchemaInfo[] schemas) in C:\projects\efcorepowertools\src\GUI\EFCorePowerTools\Handlers\ReverseEngineer\EfRevEngLauncher.cs:line 40
   at EFCorePowerTools.Handlers.ReverseEngineer.TableListBuilder.GetTableDefinitions(Boolean useEFCore5) in C:\projects\efcorepowertools\src\GUI\EFCorePowerTools\Handlers\ReverseEngineer\TableListBuilder.cs:line 39
   at EFCorePowerTools.Handlers.ReverseEngineerHandler.GetTables(String connectionString, DatabaseType databaseType, Boolean useEFCore5, SchemaInfo[] schemas) in C:\projects\efcorepowertools\src\GUI\EFCorePowerTools\Handlers\ReverseEngineerHandler.cs:line 351
   at EFCorePowerTools.Handlers.ReverseEngineerHandler.<ReverseEngineerCodeFirst>d__3.MoveNext() in C:\projects\efcorepowertools\src\GUI\EFCorePowerTools\Handlers\ReverseEngineerHandler.cs:line 98
```

**Comment by svengeance** [comment]
@ErikEJ Unfortunately it seems that it is now coercing me to use a .dacpac. If I check "Use EF 5.0 Preview", it gives me the regular timout

![image](https://user-images.githubusercontent.com/6847381/92640616-8dc44100-f2ab-11ea-90c4-6ca33b3aa72f.png)


```
System.Exception: Table list error: 
System.ArgumentException: Dacpac file not found: Data Source=<connectionstring>
   at ErikEJ.EntityFrameworkCore.SqlServer.Scaffolding.SqlServerDacpacDatabaseModelFactory.Create(String dacpacPath, DatabaseModelFactoryOptions options)
   at ReverseEngineer20.TableListBuilder.GetTableDefinitions() in C:\Code\EFCorePowerTools\src\GUI\RevEng.Core\TableListBuilder.cs:line 40
   at efreveng.Program.Main(String[] args) in C:\Code\EFCorePowerTools\src\GUI\efreveng\Program.cs:line 33

   at EFCorePowerTools.Handlers.ReverseEngineer.ResultDeserializer.BuildTableResult(String output) in C:\projects\efcorepowertools\src\GUI\EFCorePowerTools\Handlers\ReverseEngineer\ResultDeserializer.cs:line 47
   at ReverseEngineer20.ReverseEngineer.EfRevEngLauncher.GetTablesInternal(String arguments) in C:\projects\efcorepowertools\src\GUI\EFCorePowerTools\Handlers\ReverseEngineer\EfRevEngLauncher.cs:line 64
   at ReverseEngineer20.ReverseEngineer.EfRevEngLauncher.GetTables(String connectionString, DatabaseType databaseType, SchemaInfo[] schemas) in C:\projects\efcorepowertools\src\GUI\EFCorePowerTools\Handlers\ReverseEngineer\EfRevEngLauncher.cs:line 40
   at EFCorePowerTools.Handlers.ReverseEngineer.TableListBuilder.GetTableDefinitions(Boolean useEFCore5) in C:\projects\efcorepowertools\src\GUI\EFCorePowerTools\Handlers\ReverseEngineer\TableListBuilder.cs:line 39
   at EFCorePowerTools.Handlers.ReverseEngineerHandler.GetTables(String connectionString, DatabaseType databaseType, Boolean useEFCore5, SchemaInfo[] schemas) in C:\projects\efcorepowertools\src\GUI\EFCorePowerTools\Handlers\ReverseEngineerHandler.cs:line 351
   at EFCorePowerTools.Handlers.ReverseEngineerHandler.<ReverseEngineerCodeFirst>d__3.MoveNext() in C:\projects\efcorepowertools\src\GUI\EFCorePowerTools\Handlers\ReverseEngineerHandler.cs:line 98
```

```
Version 2.4.193.0

SQLite ADO.NET Provider included: 1.0.109.0
SQLite EF6 DbProvider in GAC - No

System.Data.SQLite DDEX provider - No
SQLite Simple DDEX provider - Yes
```

**Comment by ErikEJ** [comment]
OMG!

**Comment by ErikEJ** [comment]
https://github.com/ErikEJ/EFCorePowerTools/commit/9dec5223b1f1012673b009c13b7ebf639c9c5524#diff-cd9c3790e10f188151d4478cdb7e46acR85 !?

**Comment by svengeance** [comment]
I would like to subscribe to your hourly build feed. 😄

**Comment by svengeance** [comment]
Oh how I've missed you.. _note how thin that scrollbar is_

Massive success, and much faster than before when it worked as well. Thank you so much for working through this, I hope it can get into EF soon as well.

![image](https://user-images.githubusercontent.com/6847381/92642499-48edd980-f2ae-11ea-933e-598e4a8cdaed.png)

**Comment by svengeance** [comment]
Also - hopefully as a testament to the query as identical, the diff with the new scaffold reveals that the only changes are that sequences have been generated, and so have some table comments -- neither of which, I think, relate to indexes.

**Comment by ErikEJ** [comment]
@smitpatel FYI, EF Core Power Tools daily build now uses the updated GetIndexes query.

**Comment by ErikEJ** [comment]
@svengeance Yes, I think your PR should be included in EF Core 6 (if not 5.0.1)

**Comment by LynxWall** [comment]
I can confirm that build 197 is working for me as well ... it's taking about 20 seconds to load the database I'm testing against... but definitely much faster and I'm able to reverse engineer tables.

Thanks!
